q.js

```diff
@@ -1645,6 +1645,9 @@ Q["finally"] = function (object, callback) {
 
 Promise.prototype.fin = // XXX legacy
 Promise.prototype["finally"] = function (callback) {
+    if (callback === undefined) {
```

Comment:
`if (typeof callback.apply !== "function")` would throw an error for reading a property if callback were undefined, wouldn't it?